google/cloud/spanner_v1/batch.py

```diff
@@ -215,6 +216,81 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self.commit()
 
 
+class MutationGroups(_SessionWrapper):
+    """Accumulate mutations for transmission during :meth:`batch_write`.
+
+    :type session: :class:`~google.cloud.spanner_v1.session.Session`
+    :param session: the session used to perform the commit
+    """
+
+    committed = None
+
+    def __init__(self, session):
+        super(MutationGroups, self).__init__(session)
+        self._mutation_groups = []
+
+    def _check_state(self):
+        """Checks if the object's state is valid for making API requests.
+
+        :raises: :exc:`ValueError` if the object's state is invalid for making
+                 API requests.
+        """
+        if self.committed is not None:
+            raise ValueError("MutationGroups already committed")
+
+    def group(self):
+        """Returns a new mutation_group to which mutations can be added."""
+        mutation_group = BatchWriteRequest.MutationGroup()
+        self._mutation_groups.append(mutation_group)
+        return _BatchBase(self._session, mutation_group.mutations)
```

Comment:
We are returning a class that is marked as private. We should not be doing this.

Reply:
We could introduce a new public class -
```python
class MutationGroup(_BatchBase):
```

Caveat is that the client can create a standalone instance of this new class, but that should be okay.

What do you think?

Reply:
Returned a public class in https://github.com/googleapis/python-spanner/pull/1027/commits/38d9b8344e67ddab60e1a3018004a10bf77104a5